aggregate_spatial.json

```diff
@@ -0,0 +1,80 @@
+{
+    "id": "aggregate_spatial",
+    "summary": "Zonal statistics for geometries",
+    "description": "Aggregates statistics for one or more geometries (e.g. zonal statistics for polygons) over the spatial dimensions. This process passes a list of values to the reducer. In contrast, aggregate_spatial_binary() passes two values, which may be better suited for UDFs.\n\n- For **polygons**, the process considers all pixels for which the point at the pixel center intersects with the corresponding polygon (as defined in the Simple Features standard by the OGC).\n- For **points** and **lines** (line strings), the process considers all the closest pixel centers.\n\nThus, pixels may be part of multiple geometries and be part of multiple aggregations.\n\nThe data cube must have been reduced to only contain two spatial dimensions and a third dimension the values are aggregated for, for example the temporal dimension to get a time series. Otherwise this process fails with the `TooManyDimensions` error.\n\nThe number of total and valid pixels is returned together with the calculated values.",
+    "categories": [
+        "cubes",
+        "aggregate & resample"
+    ],
+    "parameters": [
+        {
+            "name": "data",
+            "description": "A raster data cube.",
+            "schema": {
+                "type": "object",
+                "subtype": "raster-cube"
+            }
+        },
+        {
+            "name": "geometries",
+            "description": "Geometries as GeoJSON on which the aggregation will be based.",
```

Comment:
Why a GeoJSON and not a vector datacube? Or both?

Reply:
We haven't defined vector cubes at all in openEO and came to the conclusion that we only use them as "black box" for the return value of this process and in no other place until we found a proper definition for us. That's a bit "hacky", but is the best we could come up with due to time constraints.

Reply:
Hm, but then you have a problem in that the output of aggregate_spatial cannot be used to chain to another aggregate_spatial, because the output types do not match... This can be quite useful if you want to extract values of multiple raster (cubes) and store them all in a single vector (cube).

Reply:
Yes, that won't be supported for now or is that something we need for the openEO use cases?

Reply:
No, it's not something we would need for the use case, but then for sure this will need to be changed in the future to make sure that the types match. Well, for that matter, the vector data cubes need to be defined for this to work altogether, because that's the return type of the process, unless we also set the return type to GeoJSON...

Reply:
Yes, there's work needed in the future, probably after the project and/or once the first use cases arrive. We handle the return type of aggregate_spatial as a special case at the moment with save_result being the only process to be able to work on it. It's basically a black box the back-end need to fill for now. This was decided to be done this was on the last calls.